--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -224,7 +224,7 @@
             MAIN_DEPENDENCY "${PROJECT_SOURCE_DIR}/DirectXTex/Shaders/CompileShaders.cmd"
             DEPENDS ${SHADER_SOURCES}
             COMMENT "Generating HLSL shaders..."
-            COMMAND ${CMAKE_COMMAND} -E env CompileShadersOutput="${COMPILED_SHADERS}" $<$<BOOL:${DIRECTX_FXC_TOOL}>:LegacyShaderCompiler=${DIRECTX_FXC_TOOL}> CompileShaders.cmd > "${COMPILED_SHADERS}/compileshaders.log"
+            COMMAND sh "${PROJECT_SOURCE_DIR}/DirectXTex/Shaders/wine-compile-shaders.sh" "${COMPILED_SHADERS}" "${DIRECTX_FXC_TOOL}"
             WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}/DirectXTex/Shaders"
             USES_TERMINAL)
     endif()
